tests/unittests/MLTest.cpp

```diff
@@ -205,6 +204,82 @@ TEST_P(MLTest, learnXor) {
   }
 }
 
+/// Learn the logarithmic function.
+TEST_P(MLTest, learnLog) {
+  unsigned numInputs = 50;
+  unsigned batchSize = 7;
+  EE_.getConfig().learningRate = 0.07;
+  EE_.getConfig().batchSize = batchSize;
+
+  auto &mod = EE_.getModule();
+  auto &PRNG = mod.getPRNG();
+  Function *F = mod.createFunction("main");
+  F->setName("learnLog");
+
+  auto *A =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "A", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+  auto *Ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "Ex", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+
+  Node *O = F->createFullyConnected("fc1", A, 5);
+  O = F->createTanh("tanh1", O);
+  O = F->createFullyConnected("fc2", O, 5);
+  O = F->createTanh("tanh2", O);
+  O = F->createFullyConnected("fc3", O, 1);
+  O = F->createTanh("tanh3", O);
+  O = F->createRegression("reg", O, Ex);
+  auto *result = F->createSave("ret", O);
+
+  // Set the training data. 
+  Tensor trainingSet(ElemKind::FloatTy, {numInputs, 1});
+  Tensor trainingLabels(ElemKind::FloatTy, {numInputs, 1});
+
+  auto TS = trainingSet.getHandle<>();
+  auto TL = trainingLabels.getHandle<>();
+
+  // Set the training date as floating number from range [0.75, 1.5).
```

Comment:
Curious, how did you select this range?

Reply:
Hi the test case takes around 2600ms.

For the range, I actually tried with a larger one in the beginning, but as the activation function shrinks all the intermediate values into the range of [-1,1]. It's hard to be trained to produce large number in the end.  It's impossible for my network to converge. 

So i chose a smaller range [0.75, 1.5], and the target output is restrained to be in between (-1,1). Making it easier to train and converge.

Reply:
Can you try getting rid of the `tanh3` activation function (right before the regression node) and seeing if that improves the ability of this net to get a larger range?  I suspect that is squashing the network's output at the final stage.

Reply:
2600ms is relatively high for a simple network like this. Let's try to play with the hyper parameters and bring this down.

Reply:
hi nadav. Sorry i might have not described clearly earlier. 

2600ms is actually the total time for running learnLog for Interpreter/MLTest, JIT/MLTest and OpenCL/MLTest.  

The OpenCl/MlTest takes the longest time, which is around 1500ms. But in other two, the running time is below 600ms.

Reply:
but still i'll try to see if i can bring the running time down. Previously i notice that the running time for an existing test "learnXor" takes the same amount of time as my test, thats why i assumed that 2600ms might be acceptable.